src/pages/es/guides/client-side-scripts.md

```diff
@@ -0,0 +1,195 @@
+```astro title="src/components/AstroHeart.astro"
+<!-- Envuelve los elementos del componente en nuestro elemento personalizado “astro-heart”. -->
+<astro-heart>
+  <button aria-label="Heart">💜</button> × <span>0</span>
+</astro-heart>
+
+<script>
+  // Define el comportamiento para nuestro nuevo tipo de elemento HTML.
+  class AstroHeart extends HTMLElement {
+    constructor() {
+			super();
+      let count = 0;
+
+      const heartButton = this.querySelector('button');
+      const countSpan = this.querySelector('span');
+
+      // Cada vez que el button es clickeado, actualiza el contador.
+			heartButton.addEventListener('click', () => {
+        count++;
+        countSpan.textContent = count;
+      });
+		}
+  }
+
+  // Dile al navegador que use nuestra clase AstroHeart para elementos <astro-heart>.
+  customElements.define('astro-heart', AstroHeart);
+</script>
+```
+
+Hay dos ventajas de usar un elemento personalizod aqui:
+
+1. En lugar de buscar en toda la pagina usando `document.querySelector()`, puedes usar `this.querySelector()`, el cual solo busca dentro de la instancia del elemento personalizado. Esto hace más fácil para trabajar  con solo los hijos de una instancia del componente a la vez.
+
+2. Aunque un `<script>` solo se ejecuta una vez, el navegador ejecutará método `constructor()` de nuestro elemento cada vez que encuentre `<astro-heart>` en la página. Esto significa que puedes escribir seguramente código para un componente a la vez, incluso si tienes la intención de utilizar este componente varias veces en una página.
+
+📚 Puedes aprender más sobre elementos personalizaods en [la guía de Componentes Web Reusables de web.dev](https://web.dev/custom-elements-v1/) y en [la introducción de MDN a elementos personalizados](https://developer.mozilla.org/es/docs/Web/Web_Components/Using_custom_elements).
+
+### Pasando variables frontmatter a scripts
+
+En componentes Astro, el codigo en [el frontmatter](/es/core-concepts/astro-components/#script-de-un-componente) entre los cercos `---` se ejecuta en el servidor y no esta disponible en el navegador. Para mandar variables desde el servidor al cliente, necesitamos una manera de guardar nuestras variables y leerlas cuando JavaScript se ejecute en el navegador.
+
+Una manera para hacer esto es usando [atributos `data-*`](https://developer.mozilla.org/es/docs/Learn/HTML/Howto/Use_data_attributes) para guardar el valor de las vairables en tu HTML. Los scripts, incluyendo elementos personalizados, pueden entonces leer estos atributos usando la propiedad de elemento `dataset` una vez que tu HTML cargue en el navegador.
+
+En este componente ejemplo, un propiedad `message` es guardad en un atributo `data-message`, para que el elemento personalizado pueda leer `this.dataset.message` y obtener el valor de la propiedad en el navegador.
+
+```astro title="src/components/AstroGreet.astro" {2} /data-message={.+}/ "this.dataset.message"
+---
+const { message = 'Welcome, world!' } = Astro.props;
```

Comment:
Maybe we could translate this example as well. I think the same phrase is at the beginning of the document:

"{ message = "¡Bienvenido, mundo!" }"